src/sentry/api/endpoints/group_suspect_releases.py

```diff
@@ -0,0 +1,91 @@
+from datetime import timedelta
+
+from django.db.models import Q
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.base import EnvironmentMixin
+from sentry.api.bases import GroupEndpoint
+from sentry.api.helpers.environments import get_environments
+from sentry.api.serializers import GroupSerializerSnuba, serialize
+from sentry.models.deploy import Deploy
+from sentry.models.grouphistory import GroupHistory, GroupHistoryStatus
+from sentry.models.release import Release
+from sentry.types.ratelimit import RateLimit, RateLimitCategory
+
+
+class GroupSuspectReleasesEndpoint(GroupEndpoint, EnvironmentMixin):
+    enforce_rate_limit = True
+    rate_limits = {
+        "GET": {
+            RateLimitCategory.IP: RateLimit(5, 1),
+            RateLimitCategory.USER: RateLimit(5, 1),
+            RateLimitCategory.ORGANIZATION: RateLimit(5, 1),
+        },
+    }
+
+    def get(self, request: Request, group) -> Response:
+        """
+        Retrieve Suspect Releases for an Issue
+        ``````````````````````````````````````
+
+        Return the suspect releases for an issue. Suspect releases are defined
+        as releases which caused the first seen issue or the latest regression
+        during the active release window (1 hour).
+        :pparam string issue_id: the ID of the issue to retrieve.
+        :auth: required
+        """
+
+        organization = group.project.organization
+        environment_ids = [e.id for e in get_environments(request, organization)]
+
+        data = serialize(group, request.user, GroupSerializerSnuba(environment_ids=environment_ids))
+
+        suspect_releases = set()
+        regression = (
+            GroupHistory.objects.filter(group=group, status=GroupHistoryStatus.REGRESSED)
+            .order_by("-date_added")
+            .first()
+        )
+
+        latest_regression_date = None
+        if regression and regression.release:
+            suspect_releases.add(regression.release)
+        else:
+            latest_regression_date = regression.date_added if regression else None
+
+        first_seen = group.first_seen
+        deploy_filter = Q(
+            organization_id=organization.id,
+            date_finished__gt=first_seen - timedelta(hours=1),
+            date_finished__lte=first_seen,
+            release__projects__in=[group.project],
+        )
+        if latest_regression_date:
+            deploy_filter |= Q(
+                organization_id=organization.id,
+                date_finished__gt=latest_regression_date - timedelta(hours=1),
+                date_finished__lte=latest_regression_date,
+                release__projects__in=[group.project],
+            )
+
+        deploys = Deploy.objects.filter(deploy_filter).select_related("release")
```

Comment:
Nit: Since you don't actually care about the deploys here, you could switch this round like
```
Release.objects.filter(projects_in=[group.project], organization_id=organization.id, deploy__date_finished__gt=...)
```
The relation name might not be `deploy`, but it'll be something like that. This allows us to avoid fetching the deploy data at all.

Not a huge deal or anything, more jfyi.

Reply:
Interesting. I couldn't find the related deploy field so leaving it as is for now.